packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingBackgroundExecutor.java

```diff
@@ -157,19 +156,10 @@ public void startBackgroundIsolate(long callbackHandle, FlutterShellArgs shellAr
                 String appBundlePath = loader.findAppBundlePath();
                 AssetManager assets = ContextHolder.getApplicationContext().getAssets();
                 if (isNotRunning()) {
-                  if (shellArgs != null) {
-                    Log.i(
-                        TAG,
-                        "Creating background FlutterEngine instance, with args: "
-                            + Arrays.toString(shellArgs.toArray()));
-                    backgroundFlutterEngine =
-                        new FlutterEngine(
-                            ContextHolder.getApplicationContext(), shellArgs.toArray());
-                  } else {
-                    Log.i(TAG, "Creating background FlutterEngine instance.");
-                    backgroundFlutterEngine =
-                        new FlutterEngine(ContextHolder.getApplicationContext());
-                  }
+                  Log.i(TAG, "Creating background FlutterEngine instance.");
+                  backgroundFlutterEngine =
+                      new FlutterEngineGroup(ContextHolder.getApplicationContext())
```

Comment:
I don't think this is going to give us what we want.  In order for you to avoid creating another isolate group and get the benefit of faster starting lower memory engines you'll have to share the FlutterEngineGroup between the plugin and the host app.  I don't think there is a way for you to do that today since there is no guarantee the host app is using the FlutterEngineGroup.

Reply:
On the flip side, isolate platform channels is about to land on stable.  This allows sending platform channel messages from background isolates.  I believe that will remove the need this plugin has for spawning an engine in the first place.  It's probably better we just lean into that.

Reply:
Thanks for the recommendation, I'll check that once it has land in stable :)